adoc/chapters/programming_interface.adoc

```diff
@@ -12761,7 +12761,9 @@ may execute a single <<command>> such as invoking a kernel, copying memory,
 or executing a host task.  It is legal for a <<command-group-scope>> to
 statically contain more than one call to a <<command>> function, but any
 single execution of the <<command-group-function-object>> may execute no more
-than one <<command>>.  The statements that call <<command, commands>> together with
+than one <<command>>. If an application fails to do this, the implementation
+must throw a synchronous [code]#exception# with the [code]#errc::invalid# error
+code. The statements that call <<command, commands>> together with
```

Comment:
Should we specify which function throws the exception?  With the current wording, it's not clear if it would be `handler::parallel_for()` or `queue::submit()`.  I think the exception should be thrown from `queue::submit()`, which is consistent with the point at which we create the requisites from accessors, and it is also consistent with the timestamp we return for `info::event_profiling::command_submit`.  If you agree, maybe word it like:

> If an application fails to do this, the function that submits the \<\<command\>\> (e.g. [code]#queue::submit#) must throw a synchronous [code]#exception# with the [code]#errc::invalid# error code.

Reply:
Makes sense, however I'm not sure about the wording. Wouldn't `queue::submit` be submitting the corresponding *command group function object*, instead of the *command* itself? Also, is there any other way to erroneously submit multiple commands, other than `queue::submit`? This can't happen for the convenience shortcut functions after all.

Reply:
> Wouldn't `queue::submit` be submitting the corresponding _command group function object_, instead of the _command_ itself?

I have the feeling this is already what you say and that sounds good.

Reply:
Yes, this wording looks good to me too.